src/Command/DonationUpdateReferenceCommand.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace AppBundle\Command;
+
+use AppBundle\Donation\DonationRequestUtils;
+use AppBundle\Entity\Donation;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Internal\Hydration\IterableResult;
+use Doctrine\ORM\QueryBuilder;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DonationUpdateReferenceCommand extends Command
+{
+    private const BATCH_SIZE = 1000;
+
+    private $em;
+    private $donationRequestUtils;
+
+    public function __construct(
+        EntityManagerInterface $em,
+        DonationRequestUtils $donationRequestUtils
+    ) {
+        $this->em = $em;
+        $this->donationRequestUtils = $donationRequestUtils;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('app:donations:update-reference')
+            ->setDescription('Update Donations reference')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $output->writeln(['', 'Starting Donations reference update.']);
+
+        $this->em->beginTransaction();
+
+        try {
+            $this->updateDonationsReference($output);
+
+            $this->em->commit();
+        } catch (\Exception $exception) {
+            $this->em->rollback();
+
+            throw $exception;
+        }
+
+        $output->writeln(['', 'Donations reference updated successfully!']);
+    }
+
+    private function updateDonationsReference(OutputInterface $output): void
+    {
+        $progressBar = new ProgressBar($output, $this->countDonations());
+
+        $count = 0;
+        foreach ($this->getDonations() as $result) {
+            $donation = reset($result);
+
+            $reference = $this->donationRequestUtils->buildDonationReference(
+                $donation->getUuid(),
+                sprintf('%s %s', $donation->getFirstName(), $donation->getLastName())
+            );
+
+            $donation->setPayboxOrderRef($reference);
+
+            $this->em->persist($donation);
```

Comment:
I don't think it's needed to call `persist` here as the object comes from doctrine ORM and thus should be already attached.